test/Microsoft.AspNetCore.Hosting.Tests/HostingApplicationTests.cs

```diff
@@ -20,16 +21,152 @@ public class HostingApplicationTests
-            var httpContextFactory = new HttpContextFactory(new DefaultObjectPoolProvider(), Options.Create(new FormOptions()), new HttpContextAccessor());
-            var hostingApplication = new HostingApplication(ctx => Task.FromResult(0), new NullScopeLogger(), new NoopDiagnosticSource(), httpContextFactory);
-            var features = new FeatureCollection();
-            features.Set<IHttpRequestFeature>(new HttpRequestFeature());
+            var hostingApplication = CreateApplication(out var features);
             var context = hostingApplication.CreateContext(features);
 
             // Act/Assert
             hostingApplication.DisposeContext(context, null);
         }
 
+#if NETCOREAPP2_0
+        [Fact]
+        public void ActivityIsAvailibleDuringBeginRequestCall()
+        {
+            var diagnosticSource = new DiagnosticListener("DummySource");
+            var hostingApplication = CreateApplication(out var features, diagnosticSource: diagnosticSource);
+
+            bool startCalled = false;
+            diagnosticSource.Subscribe(new CallbackDiagnosticListener(pair =>
+            {
+                if (pair.Key == "Microsoft.AspNetCore.Hosting.BeginRequest")
+                {
+                    startCalled = true;
+                    Assert.NotNull(Activity.Current);
+                    Assert.Equal("Microsoft.AspNetCore.Hosting.Activity", Activity.Current.OperationName);
+                }
+            }));
+
+            hostingApplication.CreateContext(features);
+            Assert.True(startCalled);
+        }
+
+        [Fact]
+        public void ActivityIsStoppedDuringEndRequestCall()
+        {
+            var diagnosticSource = new DiagnosticListener("DummySource");
+            var hostingApplication = CreateApplication(out var features, diagnosticSource: diagnosticSource);
+
+            bool endCalled = false;
+            diagnosticSource.Subscribe(new CallbackDiagnosticListener(pair =>
+            {
+                if (pair.Key == "Microsoft.AspNetCore.Hosting.EndRequest")
+                {
+                    endCalled = true;
+                    Assert.Null(Activity.Current);
+                }
+            }));
+
+            var context = hostingApplication.CreateContext(features);
+            hostingApplication.DisposeContext(context, null);
+            Assert.True(endCalled);
+        }
+
+        [Fact]
+        public void ActivityIsStoppedDuringUnhandledExceptionCall()
+        {
+            var diagnosticSource = new DiagnosticListener("DummySource");
+            var hostingApplication = CreateApplication(out var features, diagnosticSource: diagnosticSource);
+
+            bool endCalled = false;
+            diagnosticSource.Subscribe(new CallbackDiagnosticListener(pair =>
+            {
+                if (pair.Key == "Microsoft.AspNetCore.Hosting.EndRequest")
+                {
+                    endCalled = true;
+                    Assert.Null(Activity.Current);
```

Comment:
check that End event is also sent and there is a TaskStatus in the payload of End